### PR TITLE
Building the registries: duplicate content instead of conditionals.

### PR DESCRIPTION
**Pull request by themr0c**
Rewrite "building the registries": instead of trying to single-source the procedure and accumulate the conditionals, accept to duplicate some of the content.

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [x] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [x] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[id="building-an-offline-devfile-registry-image_{context}"]
+= Building an offline devfile registry image
+
+This section describes how to build an offline devfile registry image.
+Starting workspaces without relying on resources from the outside Internet requires building this image.
+The image contains all sample projects referenced in devfiles as `zip` files.
+
+.Prerequisites:
+
+* NodeJS 12.x
```
Comment (ericwill) [inline modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]: This is incorrect, and appears to be a bug in the current docs:

https://www.eclipse.org/che/docs/che-7/installation-guide/installing-che-in-a-restricted-environment/#building-offline-devfile-registry-image_che

You do not need NodeJS or yarn to build an offline devfile registry. It's needed for the plugin registry, so this just looks like a copy/paste mistake.

All that's needed is podman/docker installed. All the bits about NodeJS/yarn can be removed.

File: modules/installation-guide/partials/assembly_building-offline-registry-images.adoc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[id="building-offline-registry-images_{context}"]
+= Building offline registry images
+
+include::partial$proc_building-an-offline-devfile-registry-image.adoc[leveloffset=+1]
```
Comment (ericwill) [inline modules/installation-guide/partials/assembly_building-offline-registry-images.adoc:4]: When I deploy this, I think something is missing. The build steps for the offline plugin and devfile registries are the same, whereas one should be using NodeJS/yarn and the other is just running the `build.sh` file with no NodeJS/yarn prerequisite.

This is especially odd given the comment I left below in `modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc`

**Review by ericwill — COMMENTED** [review]
Having spent some time this week working on these articles, this is a welcome change. :smile: 

I just had a few comments about some of the offline build steps, please see my comments inline.

File: modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc (hunk 1 above)
Comment (themr0c) [reply modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]: Good!

Comment (themr0c) [inline modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]:
```suggestion
```

File: modules/installation-guide/examples/proc_che-preparing-a-private-registry.adoc [hunk 3]
```diff
@@ -27,10 +27,10 @@ Running any workspace requires these _essential_ images:
 | Images for adding plug-ins to workspaces. See xref:administration-guide:che-workspaces-architecture.adoc#che-plug-in-broker_{context}[Che plug-ins broker].
 
 | `quay.io/eclipse/che-plugin-registry` 
-| Plug-in registry that store information about plugins and the runtime containers they require. See xref:building-offline-plugin-registry-image_{context}[].
+| Plug-in registry that store information about plugins and the runtime containers they require. See xref:building-an-offline-plug-in-registry-image_{context}[].
 
 | `quay.io/eclipse/che-devfile-registry`
-| Devfile registry that store descriptions of the sample projects and how to load them as workspaces via the Getting Started tab in the dashboard. See xref:building-offline-devfile-registry-image_{context}[].
+| Devfile registry that store descriptions of the sample projects and how to load them as workspaces via the Getting Started tab in the dashboard. See xref:building-an-offline-devfile-registry-image_{context}[].
```
Comment (yhontyk) [inline modules/installation-guide/examples/proc_che-preparing-a-private-registry.adoc]:
```suggestion
| Devfile registry that stores descriptions of the sample projects and how to load them as workspaces through the Getting Started tab in the dashboard. See xref:building-an-offline-devfile-registry-image_{context}[].
```

Comment (ericwill) [reply modules/installation-guide/examples/proc_che-preparing-a-private-registry.adoc]: :+1:

File: modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc (hunk 1 above)
Comment (themr0c) [inline modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]:
```suggestion
```

File: modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+* A running version of yarn. See link:https://yarnpkg.com/getting-started/install[Installing Yarn].
+
+* `./node_modules/.bin` is in the `PATH` environment variable.
```
Comment (themr0c) [inline modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]:
```suggestion
```

File: modules/administration-guide/partials/proc_building-a-custom-devfile-registry-image.adoc [hunk 5]
```diff
@@ -1,13 +1,43 @@
...
+The image contains all sample projects referenced in devfiles.
+
+.Prerequisites
+
+* NodeJS 12.x
+
+* A running version of yarn. See: link:https://yarnpkg.com/getting-started/install[Installing Yarn].
+
+* `./node_modules/.bin` is in the `PATH` environment variable.
+
```
Comment (themr0c) [inline modules/administration-guide/partials/proc_building-a-custom-devfile-registry-image.adoc]:
```suggestion
```

Comment (themr0c) [inline modules/administration-guide/partials/proc_building-a-custom-devfile-registry-image.adoc]:
```suggestion
```

File: modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+[id="building-an-offline-devfile-registry-image_{context}"]
+= Building an offline devfile registry image
+
+This section describes how to build an offline devfile registry image.
+Starting workspaces without relying on resources from the outside Internet requires building this image.
+The image contains all sample projects referenced in devfiles as `zip` files.
+
+.Prerequisites:
+
+
```
Comment (themr0c) [inline modules/installation-guide/partials/proc_building-an-offline-devfile-registry-image.adoc]:
```suggestion
```

File: modules/installation-guide/partials/assembly_building-offline-registry-images.adoc (hunk 2 above)
Comment (themr0c) [reply modules/installation-guide/partials/assembly_building-offline-registry-images.adoc:4]: I removed the nodejs/yarm prerequisites. Is there still something to change?

**Review by yhontyk — APPROVED** [review]

File: modules/installation-guide/partials/assembly_building-offline-registry-images.adoc (hunk 2 above)
Comment (ericwill) [reply modules/installation-guide/partials/assembly_building-offline-registry-images.adoc:4]: Yep it's gone :+1:

**Review by ericwill — APPROVED** [review]
Content wise this is good, nice PR. :tada: 

There is a warning in the console:
`asciidoctor: WARNING: proc_verification-custom-registries.adoc: line 7: id assigned to block already in use: verification-custom_registries_che`

**Comment by themr0c** [comment]
@ericwill removed the id as it is never used in any xref, thanks.

File: modules/installation-guide/examples/proc_che-preparing-a-private-registry.adoc [hunk 7]
```diff
@@ -42,7 +42,7 @@ To avoid issues when only a subset of the images described in this section is ne
 
 .Plug-in sidecar images
 
-Many workspace plug-ins are run in sidecar containers to ensure their dependencies are available. The offline plug-in registry (see xref:building-offline-plugin-registry-image_{context}[]) includes a file with the list of all images referenced in its plug-ins.
+Many workspace plug-ins are run in sidecar containers to ensure their dependencies are available. The offline plug-in registry (see xref:building-an-offline-plugin-registry-image_{context}[]) includes a file with the list of all images referenced in its plug-ins.
```
Comment (themr0c) [inline modules/installation-guide/examples/proc_che-preparing-a-private-registry.adoc]:
```suggestion
Many workspace plug-ins are run in sidecar containers to ensure their dependencies are available. The offline plug-in registry (see xref:building-an-offline-plug-in-registry-image_{context}[]) includes a file with the list of all images referenced in its plug-ins.
```

File: modules/installation-guide/examples/proc_che-configuring-che-to-run-in-a-restricted-environment.adoc [hunk 8]
```diff
@@ -21,10 +21,10 @@ This procedure uses the following placeholders:
 | organization of the container-image registry
 
 | `<my-offline-plug-in-registry>`
-| offline plug-in registry, see xref:building-offline-plugin-registry-image_{context}[]
+| offline plug-in registry, see xref:building-an-offline-plugin-registry-image_{context}[]
```
Comment (themr0c) [inline modules/installation-guide/examples/proc_che-configuring-che-to-run-in-a-restricted-environment.adoc]:
```suggestion
| offline plug-in registry, see xref:building-an-offline-plug-in-registry-image_{context}[]
```

Comment (themr0c) [inline modules/installation-guide/examples/proc_che-configuring-che-to-run-in-a-restricted-environment.adoc]:
```suggestion
| offline plug-in registry, see xref:building-an-offline-plug-in-registry-image_{context}[]
```

**Pull request merged**